### PR TITLE
adds karma and jasmine Travis CI testing

### PR DESCRIPTION
**Pull request by priley86**
*introduces automated karma/jasmine testing for our compiled test pages
*adds Travis testing (for all branches/prs)
*adds Travis badge to README
*cherry-picks @bleathem js src change #341

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+node_js:
+  - '0.10'
+
+before_install:
+  - 'npm install -g bower grunt-cli'
+
+script:
+  - 'npm install'
+  - 'bower install'
+  - 'gem install jekyll -v 2.5.3'
```
Comment (bleathem) [inline .travis.yml:11]: Is the Jekyll install a leftover from a previous exploration?

**Comment by bleathem** [comment]
This looks great, I want to look through it more closely, but some initial feedback is that i feel the test SRC should be in a top level test folder, rather than in the src folder. This involves moving the contents of the current test folder into the dist folder.

File: .travis.yml (hunk 1 above)
Comment (dlabrecq) [inline .travis.yml:11]: Patternfly uses jekyll, so it's necessary to build.

**Comment by dlabrecq** [comment]
The code looks good, but wondering about the actual tests? This shows potential, but testing the chart for a legend of length 4 isn't valuable, IMO. Can the initial tests be enhanced before we decide to introduce all this infrastructure? I'm not sure we decided we wanted this?

File: .travis.yml (hunk 1 above)
Comment (priley86) [inline .travis.yml:11]: @bleathem nope, it is required for the current build...tests-src is our Jekyll templates here which get compiled to tests. I am then sourcing them as fixtures for karma/jasmine automation (in the effort to not duplicate them). I suppose their previous use was more for visual testing.

**Comment by priley86** [comment]
@bleathem so a destination structure would be: 

```
./tests/*html (jeykll templates)
./tests/js/unit/*.spec.js (karma unit tests)
./dist/tests/*html (compiled jekyll templates)
```

just want to be sure before i attempt this mapping?

**Comment by priley86** [comment]
@dlabrecq i can certainly add more tests (as we have several fixtures). Wanted to kick start the conversation first ;)

**Comment by dlabrecq** [comment]
Just wanted to see the tests you created be more useful. If we're not able to test anything valuable, then perhaps we don't need to introduce it?

BTW, can the directory structure stuff be done separate from this? I would be willing to merge Brian's PR, but let's not do those mods here.

**Comment by bleathem** [comment]
Test infrastructure / useful tests is a chicken / egg problem.  It's importatn to get the test infrastructure in place, so then we can contribute tests, and solicit test contributions from the community.

@priley86 the folder structure you proposed in your comment looks great.

I propose we keep this PR as-simple-as-possible as follows:
- [x]  Merge the PR moving the patternfly.js into the src dir
- [ ] Introduce a new PR to move the generated html test files into the dist folder
- [ ] Keep this PR focused to test infrastructure and basic tests, leaving additional test creation to subsequent PRs

**Comment by priley86** [comment]
@bleathem sounds good to me

**Comment by dtaylor113** [comment]
@bleathem sounds good to me, and this PR LGTM as well.

**Comment by dlabrecq** [comment]
I see that Brian's changes were merged separately -- that's what I was looking for.

**Comment by dtaylor113** [comment]
Added JIRA story to capture remaining work:

PatternFly base ref. GitHub repo. source code re-org - Dev
https://patternfly.atlassian.net/browse/PTNFLY-1576

**Comment by priley86** [comment]
ok - i've rebased #344 into this PR. Ready for review.

Also - @dtaylor113 helped me with JIRA/Github links! Thanks Dave!!! Noted this PR in comments here:
[PTNFLY-1550](https://patternfly.atlassian.net/browse/PTNFLY-1550)

**Pull request merged**